src/main.cpp

```diff
@@ -1,18 +1,19 @@
 #include <Arduino.h>
 
-/**
- * Just implements a basic "blink" sketch.
- * on the esp32 dev board pin D2 controls the on-board LED.
- */
-#define LED_PIN IN_LED_PIN
+//
+#if defined ENABLE_LIGHTING
+#include <lighting.h>
+#endif
 
 void setup() {
-  pinMode(LED_PIN, OUTPUT);
+#if defined ENABLE_LIGHTING
+  lighting_setup();
+#endif
 }
 
 void loop() {
-  delay(1000);
-  digitalWrite(LED_PIN, HIGH);
-  delay(1000);
-  digitalWrite(LED_PIN, LOW);
+
+#if defined ENABLE_LIGHTING
+  lighting_loop();
```

Comment:
i like this pattern too, we just might have to be careful when we implement them that they dont run any `delay`s or anything since that could maybe compound or disturb the other modules?

Reply:
Agreed, I'm assuming there is no multi threading in the processor you're proposing.  I ran into that for my xmas ornament in 2018.  I did find a lib that emulated multiple threads if we need it.

Reply:
i dont think so, although maybe making the web requests would be nice to wait in the background. not sure how exactly that works though